packages/ibm-products/src/components/Guidebanner/Guidebanner.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * Copyright IBM Corp. 2023, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// Import portions of React that are needed.
+import React, { useRef, useState } from 'react';
+
+// Other standard imports.
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import { blue90, purple70 } from '@carbon/colors';
+import {
+  CaretLeft16,
+  CaretRight16,
+  Close20,
+  Idea20,
+} from '@carbon/icons-react';
+import { Button } from 'carbon-components-react';
+import { Carousel } from '../Carousel';
+
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg } from '../../settings';
+
+// Carbon and package components we use.
+/* TODO: @import(s) of carbon components and other package components. */
+
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--guidebanner`;
+const componentName = 'Guidebanner';
+
+const defaults = {
+  collapsable: false,
+  // Labels
+  backIconDescription: 'Back',
+  closeIconDescription: 'Close',
+  collapseButtonLabel: 'Read less',
+  expandButtonLabel: 'Read more',
+  nextIconDescription: 'Next',
+};
+
+/**
+ * The guide banner sit at the top of a page, or page-level tab,
+ * to introduce foundational concepts related to the page's content.
+ *
+ * Note: all child components of GuidebannerElements must specify
+ * a 'dark' theme where applicable.
```

Comment:
Maybe I should remove this comment completely to avoid confusion. Let me explain.

The Guidebanner is always white text on a graduated background, regardless of the page's theme. The idea is that if the dev plans to include theme-enabled components such as buttons or links, they should specify their "dark" theme.

I'll update the Storybook doc to be more specific.